### ISSUE TITLE
DNeg’s XStudio on windows with CMake

### ISSUE THREAD

**Issue by giuseppeICH** [AcademySoftwareFoundation/xstudio#100]
Hi,

i am trying to install DNeg’s XStudio on windows with CMake, but i am finding many difficulties.
Every time I do the CMake setup it tells me that a new package is missing.

the first one was “nlohmann_json”
(solved by downloading the package from GitHub and placed in the folder
./XStudio/xstudio/build/vcpkg/packages/nlohmann-json_x64-windows/share/nlohmann_json)

the second one “pybind11”
(solved by downloading the package via Python and placed in the folder
C:/Users/User/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0/LocalCache/local-packages/Python312/site-packages/pybind11/share/cmake/pybind11

now it is “cafConfig.cmake” which i can’t figure out where to find it.

I wanted to know if there was a single guide on how to implement XStudio, because online I can’t find a comprehensive guide on the various procedures to follow.

the only generic guide is on

[mpkepic/xstudio/blob/windows/docs/build_guides/windows.md](https://github.com/mpkepic/xstudio/blob/windows/docs/build_guides/windows.md)
but it is missing all the contingencies that I am finding.

Thanks in advance

Giuseppe

**Comment by fjallraven** [comment]
Try to enable USE_VCPKG, the windows branch has a vcpkg.json file which should make dependency install alot easier.

**Comment by ndeebook** [comment]
Hello here

I have a similar experience:
- first, a hash problem with openimageio\portfile.cmake
- then, ensurepip problem (https://discourse.cmake.org/t/cmake-error-failed-to-ensurepip/10019)
- after, it's pybind11 not found
- spdlog
- ... and so on

Enabling USE_VCPKG does not help :/

**Comment by fjallraven** [comment]
Sounds like installing and invoking vcpkg failed, would you be able to share beginning on cmake log?

You are on mpkepic/xstudio windows branch?

**Comment by xXTR0YXx** [comment]
I was manually downloading a lot of dependencies using vcpkg now and manually edited all the pathnames, but currently stuck with liblzma which is installed, but cmake is looking for a lzma_auto_decoder which is not part of the installation and also nowhere to be found

**Comment by mpkepic** [comment]
Hi!  I've done a significant overhaul on the Cmake/vcpkg integration and cmake install target support among other things.  I recommend checking out the latest commit on the pull request.

Note it now supports Visual Studio 2022, comes with some useful presets to make configuring easier, and has a bunch of build reliability improvements.

**Comment by xXTR0YXx** [comment]
yes, you posted an update on the next day, and I could finally make it work. It was also crucial to run the power-shell script setting up the environment properly which doesn't work with an unaltered system where running scripts is not allowed. it is now up and running, still get some python related error messages on boot-up and xStudio doesn't recognize EXR sequences as such but instead only loads one image

**Comment by mpkepic** [comment]
Thanks, I'm spinning off that issue into #106; see there for updates.

**Comment by mpkepic** [comment]
@xXTR0YXx, I've closed out #106, if you want to pull and try it again it should be resolved.

**Comment by xXTR0YXx** [comment]
Thanks..will look into that today. I have also the issue that loading from my NAS is not supported, but I have seen that this is mentioned somewhere already. Guess it is the double-slash issue if I remember right

**Comment by mpkepic** [comment]
> Thanks..will look into that today. I have also the issue that loading from my NAS is not supported, but I have seen that this is mentioned somewhere already. Guess it is the double-slash issue if I remember right


Can you please open a new issue for this with any sort of useful examples?  I'm trying (and failing) to prevent these massive monolithic issues that keep morphing :)

**Comment by xXTR0YXx** [comment]
Will do. But currently I wasn't able to build your fork due to VCPKG not being able to create MP3LAME

**Comment by mpkepic** [comment]
can you provide the relevant section from the build logs? alternatively you can use a prebuilt ffmpeg 5.1.2 and set FFMPEG_ROOT

**Comment by xXTR0YXx** [comment]
Also gives the error when using vcpkg install directly and when using "full" vcpkg without manifest using vcpkg install mp3lame


[build-x64-windows-rel-out.log](https://github.com/user-attachments/files/15774870/build-x64-windows-rel-out.log)

**Comment by mpkepic** [comment]
> I used the prebuild FFMPEG on my first build, but adding FFMPEG_ROOT set to C:\ffmpeg\bin seems to be ignored

If using prebuilt, you can remove ffmpeg from the manifest, add the FFMPEG_ROOT and point it to the non-bin directory; make sure your ffmpeg prebuilt is the full shared version (with dll and libs), not just the executable.

**Comment by mpkepic** [comment]
Given the error is failing to find some very core files, validate two things for me:
1) have you enabled long file paths (https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry)
2) do you have disk space available on your drive (and did you when you installed visual studio)?

Alternatively, you can be more discerning about which packages you pull in by modifying the manifest to set the features in ffmpeg and only include specific features.
https://vcpkg.link/ports/ffmpeg

**Comment by xXTR0YXx** [comment]
yes, set the file path limitation from 0 to 1 and have plenty of drive space on each drive on this machine.

So just to make sure, the manifest is vcpkg.json?

**Comment by mpkepic** [comment]
Yes, vcpkg.json is the manifest.

**Comment by xXTR0YXx** [comment]
There is not much I can delete from the manifest regarding ffmpeg. The upper part seems to be part of the opencolorio dependency and can't be delete. The lower part I can delete, but doesn't make a difference when trying to build.

![manifest](https://github.com/AcademySoftwareFoundation/xstudio/assets/125904224/fd20088a-a392-492e-acdc-c9603e0dac2c)

**Comment by mpkepic** [comment]
What I meant is you can modify the "features": ["all"] to be, instead of "all", a list of features you wish to include, as enumerated here:
https://vcpkg.link/ports/ffmpeg
As an example, you could enumerate every subpackage from `all` except mp3lame.

As far as why mp3lame won't work, 🤷 but given it fails with vcpkg outside of the xstudio build, I suspect it is something to do with your specific computer configuration.

**Comment by tedwaine** [comment]
Hi - closing this issue.

Windows cmake build files have been heavily reworked and tested for the current version of xSTUDIO. Please look at the new build documentation, give it a try. Open a new issue or talk to us on Academy Software Foundation slack channel #ori-xstudio-discussion if you have problems.
